### DISCUSSION TITLE
Editor. Add posibility to lock [slider length on timing line] and [slider shape].

### DISCUSSION THREAD

**Discussion by okolenmi** [ppy/osu#22196] [Ideas]
It's really uncofomfortable to balance between timing length and length of slider to adjust it to certain shape. It's also a pain when you need to change slider speed and then timing length changes as well.
What if we will lock slider timing or shape? In this case we have:
- an extra possibility to freely transform slider as you want - its speed will be calculated automaticly _(slider length locked)_
- auto slider scaling (proportional zoom in/out) when you change slider speed manually _(slider length locked)_
- auto slider speed scaling after timing length change _(slider shape locked)_ 

![sll](https://user-images.githubusercontent.com/55319814/212448404-8b2c6d73-88be-4512-9e7b-de2f10fcd541.png)

![ssl](https://user-images.githubusercontent.com/55319814/212448413-9d17bb3a-9db2-40d5-8af4-2df0dc2d6b46.png)

**Comment by peppy** [comment]
Are you sure you don't just want to shift-drag the end of slider in the timeline..?

**Comment by OliBomby** [comment]
> It's really uncofomfortable to balance between timing length and length of slider to adjust it to certain shape.

I think this is a suitable solution https://github.com/ppy/osu/pull/25953
